src/librustdoc/json/conversions.rs

```diff
@@ -170,12 +170,10 @@ impl FromWithTcx<clean::TypeBindingKind> for TypeBindingKind {
     }
 }
 
-crate fn from_def_id(did: FakeDefId) -> Id {
+crate fn from_item_id(did: ItemId) -> Id {
     match did {
-        FakeDefId::Real(did) => Id(format!("{}:{}", did.krate.as_u32(), u32::from(did.index))),
-        // We need to differentiate real and fake ids, because the indices might overlap for fake
-        // and real DefId's, which would cause two different Id's treated as they were the same.
-        FakeDefId::Fake(idx, krate) => Id(format!("F{}:{}", krate.as_u32(), u32::from(idx))),
+        ItemId::DefId(did) => Id(format!("{}:{}", did.krate.as_u32(), u32::from(did.index))),
+        _ => todo!("how should json ItemId's be represented?"),
```

Comment:
This might be a good way to handle the JSON ices actually - cc @CraftSpider @aDotInTheVoid

Reply:
This could easily fix blanket impls I imagine, as now we can identify them more easily. However, I think re-exports would still be broken... Hmm.

Reply:
The json conversion is the last thing that is missing before this PR can be merged. How should the new ID's be represented?

Reply:
I don't have a perfect representation idea right now, but I think for now, maybe try auto as `a:{}:{}-{}:{}`, blanket as `b:{}:{}-{}:{}`, and primitive as `p:{}`? I'll have to think more on if I can come up with better ideas, but for now that would at least be 'information preserving' per se.